Assets/Scripts/Tools/ToolPanelController.cs

```diff
@@ -4,11 +4,13 @@ namespace Assets.Scripts.Tools
 {
   public class ToolPanelController : MonoBehaviour, IToolChangeObserver
   {
-    public TextMesh gunModeText;
+    public TextMesh toolgunModeText;
     public Renderer panel;
 
-    public void OnToolsChanged(ITool newTool) => gunModeText.text = newTool.ToolName;
+    public void OnToolsChanged(ITool newTool) => SetToolgunModeText(newTool.ToolName);
 
     public void SetBackgroundColor(Color color) => panel.material.color = color;
+
+    public void SetToolgunModeText(string text) => toolgunModeText.text = text;
```

Comment:
In the future I would like it to be changed so that ToolPanelController observes some global state and changes accordingly/decides what the text should be.
Now this controller doesn't control anything, rather is being controlled by everything else. That makes the information flow convoluted and prone to errors/inconsistency

Reply:
you are probably right but I don't know exactly where the state of all tools should be stored at once
and I think that adding new tool would not be as easy as it is now